### PR TITLE
Add pagination data to the /api/deltas/date API.

### PR DESCRIPTION
**Pull request by garionpin**
Add pagination data and refactor response type for /api/deltas/date API
endpoint.

### CODE REVIEW

File: webapp/src/main/java/com/box/l10n/mojito/service/tm/TMTextUnitVariantRepository.java [hunk 1]
```diff
@@ -42,11 +42,12 @@ List<TMTextUnitVariant> findByTmTextUnitTmRepositoriesAndLocale_Bcp47TagNotOrder
    */
   @Query(
       value =
-          "select distinct tuv.id, "
-              + "   tu.name as tmTextUnitName, "
-              + "   tuv.locale.bcp47Tag as bcp47Tag,"
-              + "   tuv.content as content, "
-              + "   'UNKNOWN' as deltaType "
+          "select distinct new com.box.l10n.mojito.service.tm.TextUnitVariantDeltaDTO("
```
Comment (aurambaj) [inline webapp/src/main/java/com/box/l10n/mojito/service/tm/TMTextUnitVariantRepository.java:45]: it was working without an actual DTO instance before? just wondering why the change?

Comment (garionpin) [reply webapp/src/main/java/com/box/l10n/mojito/service/tm/TMTextUnitVariantRepository.java:45]: Needed a DTO for the return type of the api/delta/date API anyways and didn't think an extra conversion was warranted.
